synapse/handlers/message.py

```diff
@@ -682,7 +684,10 @@ async def send_nonmember_event(
                     event.event_id,
                     prev_event.event_id,
                 )
-                return await self.store.get_stream_id_for_event(prev_event.event_id)
+                return (
+                    prev_event,
+                    await self.store.get_stream_id_for_event(prev_event.event_id),
```

Comment:
I see that below we use `event.internal_metadata.stream_ordering` can something similar be used here?

Reply:
Oh, yes.